### PR TITLE
[v6r16] DynamicMonitoring task in SysAdmin

### PR DESCRIPTION
**Pull request by Sbalbp**
Added a periodic task to the SysAdmin service (can be enabled from CS by setting a flag, just like the pseudostatic monitoring) that retrieves profiling information from running components and sends it to a RabbitMQ queue for later storage in ElasticSearch.
Requires PR's #2619 and #2827 to be merged before.

### CODE REVIEW

File: FrameworkSystem/Service/SystemAdministratorHandler.py [hunk 1]
```diff
@@ -43,6 +45,33 @@ def initializeHandler( cls, serviceInfo ):
...
+      sysSetup = result[ 'Value' ]
+      result = gConfig.getOptionsDict( 'Systems/%s/%s/MessageQueueing/%s' % ( system, sysSetup, queueName ) )
+      if not result[ 'OK' ]:
+        return result
+      parameters = result[ 'Value' ]
+
+      # Setup the RabbitMQ connection with the retrieved parameters
+      result = SystemAdministratorHandler.rabbitMQ.setupConnection( 'Framework', 'ComponentMonitoring', parameters, False )
+      if not result[ 'OK' ]:
+        gLogger.error( result[ 'Message' ] )
```
Comment (chaen) [inline FrameworkSystem/Service/SystemAdministratorHandler.py]: don't you miss a return statement here ?

**Comment by fstagni** [comment]
I think also SysAdmin would benefit from having an Integration test in DIRAC/tests/Integration/Framework

File: FrameworkSystem/Service/SystemAdministratorHandler.py [hunk 2]
```diff
@@ -43,6 +45,37 @@ def initializeHandler( cls, serviceInfo ):
...
+        return result
+      parameters = result[ 'Value' ]
+
+      # Setup the RabbitMQ connection with the retrieved parameters
+      result = SystemAdministratorHandler.rabbitMQ.setupConnection( 'Framework', 'ComponentMonitoring', parameters, False )
+      if not result[ 'OK' ]:
+        gLogger.error( 'DynamicMonitoring couldn\'t be enabled: %s' % result[ 'Message' ] )
+        return result
+
+      gThreadScheduler.addPeriodicTask( 120, client.storeProfiling )
```
Comment (zmathe) [inline FrameworkSystem/Service/SystemAdministratorHandler.py]: We have to decide what to put here. I propose to double at least.

File: FrameworkSystem/Service/SystemAdministratorHandler.py [hunk 3]
```diff
@@ -651,3 +684,42 @@ def export_storeHostInfo( self ):
...
+      gLogger.error( result[ 'Message' ] )
+      return S_ERROR( result[ 'Message' ] )
+    setupComps = result[ 'Value' ]
+
+    # Get the profiling information for every running component and send it to RabbitMQ
+    for cType in setupComps:
+      for system in setupComps[ cType ]:
+        for comp in setupComps[ cType ][ system ]:
+          pid = startupComps[ '%s_%s' % ( system, comp ) ][ 'PID' ]
+          profiler = Profiler.Profiler( pid )
```
Comment (zmathe) [inline FrameworkSystem/Service/SystemAdministratorHandler.py]: getStartupComponentStatus also return CPU, Memory,... usage. Profiler much better maybe we can drop that functionality in getStartupComponentStatus method and use profiler here as well.

File: FrameworkSystem/Service/SystemAdministratorHandler.py [hunk 4]
```diff
@@ -43,6 +45,37 @@ def initializeHandler( cls, serviceInfo ):
...
+    if dynamicMonitoring:
+      client = SystemAdministratorClient( 'localhost' )
+
+      # Get the RabbitMQ parameters from the CS
+      result = gConfig.getOption( 'DIRAC/Setup' )
+      if not result[ 'OK' ]:
+        gLogger.error( 'DynamicMonitoring couldn\'t be enabled: %s' % result[ 'Message' ] )
+        return result
+      setup = result[ 'Value' ]
+      result = gConfig.getOption( 'DIRAC/Setups/%s/%s' % ( setup, system ) )
```
Comment (andresailer) [inline FrameworkSystem/Service/SystemAdministratorHandler.py]: Where is the variable  ```system``` declared?

**Comment by zmathe** [comment]
It is recreated: #2970

**Pull request closed without merging**